primitives/runtime/src/offchain/storage.rs

```diff
@@ -58,30 +78,40 @@ impl<'a> StorageValueRef<'a> {
 	/// Retrieve & decode the value from storage.
 	///
 	/// Note that if you want to do some checks based on the value
-	/// and write changes after that you should rather be using `mutate`.
+	/// and write changes after that, you should rather be using `mutate`.
 	///
-	/// The function returns `None` if the value was not found in storage,
-	/// otherwise a decoding of the value to requested type.
-	pub fn get<T: codec::Decode>(&self) -> Option<Option<T>> {
+	/// Returns the value if stored.
+	/// Returns an error if the value could not be decoded.
+	pub fn get<T: codec::Decode>(&self) -> Result<Option<T>, StorageRetrivalError> {
```

Comment:
I'd still prefer to avoid the `Option` layer. If nothing else, `x.is_ok()` is nicer to use than `matches!(x, Ok(Some(_)))`. 

I understand that in general it is not an error in the sense of an abnormal condition if a value is not found in storage. However, that's IMO a weak argument for returning `Result<Option<T>, _>`. The point of having an error enum is to let downstream users distinguish between different reasons the function didn't complete successfully. If we didn't care about that, we'd just always return `Option<T>`. Given that we have a `Result` in the first place, it's nicer for the user to unpack just one layer vs two.

Reply:
It is easier to use that way but it is a little unexpected to find that None was one of the Error choices. I think the changes are strictly an improvement on what was there before and should have less cognative load than Option<Option<>>.

Reply:
On a higher-level dealing with a storage error will probably be different than dealing with a value not being found, e.g. in most cases you can deal with the later. If you isolate the actual errors in result then you can just use `?` and then have some higher-level logic that deals with missing values (which may be expected). IMO we should keep the interface as `Result<Option<_>, _>`.